### PR TITLE
feat: DelComment

### PR DESCRIPTION
**Pull request by NaturalGao**
new DelComment function

# PR Details

## Description

增加删除批注功能函数 DelComment [#849 ]

## Motivation and Context

最近项目有用到这个库，挺棒的一个库！刚好在社群又看到有这么个需求，于是尝试贡献下自己的代码。

## How Has This Been Tested

添加了一个单元测试函数

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: comment.go [hunk 1]
```diff
@@ -140,6 +141,35 @@ func (f *File) AddComment(sheet, cell, format string) error {
 	return err
 }
 
+// DelComment provides the method to delete comment in a sheet by given worksheet
+//
+//	err := f.DelComment("Sheet1", "A30")
+func (f *File) DelComment(sheet, cell string) (err error) {
+	sheetFile, sheetExist := f.sheetMap[sheet]
```
Comment (xuri) [inline comment.go:148]: Use the `workSheetReader` to check the worksheet if exists, and you will also get the pointer of the deserialized worksheet structure. This pull request won't remove the `legacyDrawing` of the worksheet when all comments on the worksheet have been removing. I think we need to remove it, so receive the pointer of the deserialized worksheet structure here to do that, and you needn't create a new error for not exist worksheet. Note that the exported errors should be defined in the `errors.go`.

File: comment.go [hunk 2]
```diff
@@ -140,6 +141,35 @@ func (f *File) AddComment(sheet, cell, format string) error {
...
+
+	if !sheetExist {
+		err = errors.New(sheet + "is not exist")
+		return
+	}
+
+	target := f.getSheetComments(filepath.Base(sheetFile))
+
+	if target == "" {
+		err = errors.New("comment target is null")
```
Comment (xuri) [inline comment.go:158]: This internal error can be skipped, remove the errors return.

File: comment.go [hunk 3]
```diff
@@ -298,6 +328,27 @@ func (f *File) addComment(commentsXML, cell string, formatSet *formatComment) {
...
+		for k, v := range comments.CommentList.Comment {
+			if v.Ref == cell {
+				l := len(comments.CommentList.Comment)
+				if l > 1 {
+					comments.CommentList.Comment = append(
+						comments.CommentList.Comment[:k],
+						comments.CommentList.Comment[k+1:]...,
+					)
+				} else {
+					comments.CommentList.Comment = []xlsxComment{}
```
Comment (xuri) [inline comment.go:343]: The worksheet relationships and the `legacyDrawing` of the worksheet also should be removed, if all comments on the worksheet have been removed.

File: comment.go [hunk 4]
```diff
@@ -298,6 +328,27 @@ func (f *File) addComment(commentsXML, cell string, formatSet *formatComment) {
...
+	if comments := f.commentsReader(path); comments != nil {
+		for k, v := range comments.CommentList.Comment {
+			if v.Ref == cell {
+				l := len(comments.CommentList.Comment)
+				if l > 1 {
+					comments.CommentList.Comment = append(
+						comments.CommentList.Comment[:k],
+						comments.CommentList.Comment[k+1:]...,
+					)
+				} else {
```
Comment (xuri) [inline comment.go:342]: In my personal approach of choosing one is that using continue statement instead of else.

File: comment_test.go [hunk 5]
```diff
@@ -46,6 +46,23 @@ func TestAddComments(t *testing.T) {
 	assert.EqualValues(t, len(NewFile().GetComments()), 0)
 }
 
+func TestDelComments(t *testing.T) {
```
Comment (xuri) [inline comment_test.go:49]: Rename `TestDelComments` to `TestDeleteComment`.

**Review by xuri — CHANGES_REQUESTED** [review]
Thanks for your PR. I've left some comments.

**Review by xuri — APPROVED** [review]
I made some changes based on your branch.

**Comment by NaturalGao** [comment]
happy weekend! Thank you for your help! @xuri

**Pull request merged**